python/pyspark/sql/utils.py

```diff
@@ -282,3 +284,41 @@ def get_dataframe_class() -> Type["DataFrame"]:
         return ConnectDataFrame  # type: ignore[return-value]
     else:
         return PySparkDataFrame
+
+def isinstance_iterable(obj: Any) -> bool:
+    """
+    Check for iterability (with string-safe and Column-safe modifications).
+    """
+    if isinstance(obj, str):
+        return False
+    try:
+        iter(obj)
+        return True
+    except (TypeError, PySparkTypeError):
```

Comment:
PySparkTypeError is a TypeError -- should this be removed due to redundancy, or should the dependency be kept explicit?